### ISSUE TITLE
Hugo does not load changed i18n string after detecting changes in /i18n

### ISSUE THREAD

**Issue by gabrielmaldi** [gohugoio/hugo#7202]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
Hugo Static Site Generator v0.79.1-EDB9248D/extended darwin/amd64 BuildDate: 2020-12-19T15:48:14Z
</pre>

### Does this issue reproduce with the latest release?
Yes

---

When running `hugo server` (both with and without `--disableFastRender`), if I change any translations file in `/i18n`, the change isn't reflected in the generated site, even after refreshing the browser myself. If I kill `hugo server` and run it again, it does reflect the modified text.

Interestingly, `hugo server` does watch the `/i18n` folder and correctly picks up changes because it prints `i18n changed "/i18n/es.toml": WRITE` and (if run without `--disableLiveReload`) it even refreshes the page. But the modified text is not applied.

I also tried using these flags with no success: `--ignoreCache --gc --noHTTPCache --forceSyncStatic --cleanDestinationDir`.

More info. in case this is a corner case related to my (maybe particular) setup:

`/layouts/_default/single.html`
```go-html-template
{{ define "main" }}

{{ if fileExists (printf "layouts/partials/%s/%s.html" .Lang .File.BaseFileName) }}
  {{ partial (printf "%s/%s" .Lang .File.BaseFileName) . }}
{{ else }}
  {{ partial .File.BaseFileName . }}
{{ end }}

{{ end }}
```

`/layouts/partials/my-page.html`
```go-html-template
{{ $js := resources.Get "js/my-script.js" | resources.ExecuteAsTemplate (printf "js/my-script-%s.js" .Lang) . }}
{{ if $.Site.Params.Minify }}
  {{ $js = $js | minify | fingerprint "sha512" }}
{{ end }}
<script defer src="{{ $js.RelPermalink }}" integrity="{{ $js.Data.Integrity }}"></script>
```

`/assets/js/my-script.js`
```js
var text = "{{ i18n `page_myTranslation` }}",
```

`/i18n/es.toml`
```toml
page_myTranslation.other = "Mi traducción"
```

Thanks!

**Comment by gabrielmaldi** [comment]
This is a bug and still repros on `Hugo Static Site Generator v0.74.3/extended darwin/amd64 BuildDate: unknown`

**Comment by gabrielmaldi** [comment]
This bug still repros in `Hugo Static Site Generator v0.79.1-EDB9248D/extended darwin/amd64 BuildDate: 2020-12-19T15:48:14Z`

**Comment by davidsneighbour** [comment]
Can you please rename this issue into "Hugo does not load changed i18n string after detecting changes in /i18n" to make more clear in the subject, that it rebuilds, but not using the changed information?

Some further triage: 

- are all your i18n labels english (a-z, no special characters)
- how does your configuration section for languages look like?

**Comment by gabrielmaldi** [comment]
@davidsneighbour, thanks for looking into this!

Thanks to your questions I investigated further and solved the issue.

I was calling `ExecuteAsTemplate` like this:

```go-html-template
{{ $js := (resources.Get "js/shop.js" | resources.ExecuteAsTemplate (printf "js/shop-%s.js" .Lang) .) }}
```

And since `ExecuteAsTemplate` is cached and uses the path as the key, the file wasn't being regenerated.

I changed that line to:

```go-html-template
{{ $js := (resources.Get "js/shop.js" | resources.ExecuteAsTemplate (printf "js/shop-%s-%s.js" .Lang now.Unix) .) }}
```

And now everything works as expected 🥳.

This is the [second time](https://github.com/gohugoio/hugo/issues/7539) I got bitten by `ExecuteAsTemplate`; maybe something should be added to the docs?

Thanks!
